extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/job/PositionalJob.java

```diff
@@ -7,22 +7,49 @@
  */
 package org.eclipse.smarthome.binding.astro.internal.job;
 
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.eclipse.smarthome.binding.astro.internal.job.Job.checkNull;
+
 import org.eclipse.smarthome.binding.astro.handler.AstroThingHandler;
 import org.eclipse.smarthome.binding.astro.internal.AstroHandlerFactory;
-import org.quartz.JobDataMap;
 
 /**
- * Calculates and publishes astro positional data.
+ * Scheduled Job for Planet Positions
  *
  * @author Gerhard Riegler - Initial contribution
+ * @author Amit Kumar Mondal - Implementation to be compliant with ESH Scheduler
  */
-public class PositionalJob extends AbstractBaseJob {
+public final class PositionalJob implements Job {
+
+    private final String thingUID;
+
+    /**
+     * Constructor
+     *
+     * @param thingUID
+     *            thing UID
+     * @throws NullPointerException
+     *             if the provided argument is {@code null}
+     */
+    public PositionalJob(String thingUID) {
+        checkNotNull(thingUID, "Thing UID cannot be null");
+        this.thingUID = thingUID;
+    }
 
+    /** {@inheritDoc} */
     @Override
-    protected void executeJob(String thingUid, JobDataMap jobDataMap) {
-        AstroThingHandler astroHandler = AstroHandlerFactory.getHandler(thingUid);
-        if (astroHandler != null) {
-            astroHandler.publishPositionalInfo();
+    public void run() {
+        AstroThingHandler astroHandler = AstroHandlerFactory.getHandler(thingUID);
+        if (checkNull(astroHandler, "AstroThingHandler is null")) {
+            return;
         }
+        astroHandler.publishPositionalInfo();
```

Comment:
If you call this method you will run into a `Nullpointer` [here](https://github.com/eclipse/smarthome/blob/master/extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/calc/SunCalc.java#L75)

This is because `sun` has not been initialized yet. The line which initializes the `sun` for the `SunHandler` is [here](https://github.com/eclipse/smarthome/blob/master/extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/handler/SunHandler.java#L44). So before this PR this was working because `publishPositionalInfo()` was called by `publishDailyInfo()`. The method `publishPositionalInfo()` itself does not initialize `sun`.

A solution would be to initialize `sun` in the constructor of the `SunHandler` and not inside any method.